src/ReportGenerator.Core/Reporting/Builders/TextDeltaSummaryReportBuilder.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Palmmedia.ReportGenerator.Core.Common;
+using Palmmedia.ReportGenerator.Core.Logging;
+using Palmmedia.ReportGenerator.Core.Parser.Analysis;
+using Palmmedia.ReportGenerator.Core.Properties;
+
+namespace Palmmedia.ReportGenerator.Core.Reporting.Builders
+{
+    /// <summary>
+    /// Creates summary report in Text format (no reports for classes are generated).
+    /// Shows difference in coverage between the current and latest history run.
+    /// </summary>
+    public class TextDeltaSummaryReportBuilder : IReportBuilder
+    {
+        /// <summary>
+        /// The Logger.
+        /// </summary>
+        private static readonly ILogger Logger = LoggerFactory.GetLogger(typeof(TextSummaryReportBuilder));
+
+        /// <summary>
+        /// Gets the report type.
+        /// </summary>
+        /// <value>
+        /// The report type.
+        /// </value>
+        public string ReportType => "TextDeltaSummary";
+
+        /// <summary>
+        /// Gets or sets the report context.
+        /// </summary>
+        /// <value>
+        /// The report context.
+        /// </value>
+        public IReportContext ReportContext { get; set; }
+
+        /// <summary>
+        /// Creates a class report.
+        /// </summary>
+        /// <param name="class">The class.</param>
+        /// <param name="fileAnalyses">The file analyses that correspond to the class.</param>
+        public void CreateClassReport(Class @class, IEnumerable<FileAnalysis> fileAnalyses)
+        {
+        }
+
+        /// <summary>
+        /// Creates the summary report.
+        /// </summary>
+        /// <param name="summaryResult">The summary result.</param>
+        public void CreateSummaryReport(SummaryResult summaryResult)
+        {
+            if (summaryResult == null)
+            {
+                throw new ArgumentNullException(nameof(summaryResult));
+            }
+
+            HistoricCoverage[] historicCoverages = HistoricCoverages.GetOverallHistoricCoverages(this.ReportContext.OverallHistoricCoverages)
+                                                                    .TakeLast(2)
+                                                                    .ToArray();
+            if (historicCoverages.Length != 2)
+            {
+                throw new InvalidOperationException("No historic coverage to comapare.");
+            }
+
+            string targetDirectory = this.ReportContext.ReportConfiguration.TargetDirectory;
```

Comment:
this directory creation, could also be adapted into a method. this would make it easier to read and separate responsibility from your method.

private string CreateDirectory()
{
            string targetDirectory = this.ReportContext.ReportConfiguration.TargetDirectory;
            if (this.ReportContext.Settings.CreateSubdirectoryForAllReportTypes)
            {
                targetDirectory = Path.Combine(targetDirectory, this.ReportType);

                if (!Directory.Exists(targetDirectory))
                {
                    try
                    {
                        Directory.CreateDirectory(targetDirectory);
                    }
                    catch (Exception ex)
                    {
                        Logger.ErrorFormat(Resources.TargetDirectoryCouldNotBeCreated, targetDirectory, ex.GetExceptionMessageForDisplay());
                        return ""
                    }
                }
            }
            return targetDirectory;
}